File: easy/3314_construct_the_minimum_bitwise_array_i/solution.py
from typing import List


class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        bitwise_array = []

        for i in range(len(nums)):
            for j in range(nums[i]):
                if j | (j + 1) == nums[i]:
                    bitwise_array[i] = j
                    break

                j += 1
            else:
                bitwise_array[i] = -1

        return bitwise_array

File: easy/3314_construct_the_minimum_bitwise_array_i/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<int> minBitwiseArray(std::vector<int> &nums) {
    std::vector<int> bitwiseArray;
    bitwiseArray.reserve(nums.size());

    for (const int &num : nums) {
      if (num & 1)
        bitwiseArray.push_back(num & ~(((num + 1) & ~num) >> 1));
      else
        bitwiseArray.push_back(-1);
    }

    return bitwiseArray;
  }
};
